editoast/src/views/infra/mod.rs

```diff
@@ -727,12 +738,26 @@ async fn load(
     Ok(StatusCode::NO_CONTENT)
 }
 
-/// Builds a Core cache_status request, runs it
-pub async fn fetch_infra_state(_infra_id: i64, _core: &CoreClient) -> Result<InfraStateResponse> {
+#[derive(Debug, Error, EditoastError)]
+#[editoast_error(base_id = "infra_state")]
+pub enum InfraStateError {
+    #[error("Failed to fetch infra state: {0}")]
+    #[editoast_error(status = 500)]
+    FetchError(#[from] editoast_osrdyne_client::Error),
+}
+
+pub async fn fetch_infra_state(
+    infra_id: i64,
+    osrdyne: &OsrdyneClient,
+) -> Result<InfraStateResponse> {
```

Comment:
`InfraStateResponse` is a `core_client` definition, which is not relevant anymore. You can delete `src/core/infra_state.rs` and return directly the `InfraState`.